tmf/org.eclipse.tracecompass.tmf.ui/src/org/eclipse/tracecompass/tmf/ui/widgets/timegraph/widgets/TimeGraphControl.java

```diff
@@ -2333,8 +2345,23 @@ protected void drawMarker(IMarkerEvent marker, Rectangle bounds, ITimeDataProvid
             case SymbolType.CIRCLE:
                 SymbolHelper.drawCircle(gc, color, symbolSize, rect.x, y);
                 break;
-            default:
+            case SymbolType.DIAMOND:
                 SymbolHelper.drawDiamond(gc, color, symbolSize, rect.x, y);
+                break;
+            default:
+                char[] ca = symbol.toCharArray();
+                int first = ca[0];
+                /* check if the string is of length 1 or if the first character is an emoji */
+                if (ca.length == 1 || (first >= EMOJI_LOW && first <= EMOJI_HIGH)) {
```

Comment:
I looked at this again. This limits what can be drawn to single letters or numbers _or_ any number of emojis. What if we want to draw a normal double- or triple-digit number? What about just checking if the symbol type value is (not) `SymbolType.NONE`:

```suggestion
                /* draw symbol type value directly if not SymbolType.NONE */
                if (!symbol.equals(SymbolType.NONE)) {
```

Reply:
> One thing I noticed is that an emoji seems to be drawn slightly bigger (specifically taller) than a normal character (same height factor of 1.0f):

I think that's because the emoji takes the full height given, but fonts keep some height for ascenders and descenders?

> I looked at this again. This limits what can be drawn to single letters or numbers or any number of emojis. What if we want to draw a normal double- or triple-digit number? What about just checking if the symbol type value is (not) SymbolType.NONE:

I agree, you can have any length string output as long as the first character is an emoji now. So I think it's ok to accept any length string as the symbol type (except for the reserved values e.g. "square"...).

Then there would be no case for the DIAMOND default. Please update the code and Javadoc accordingly.

Reply:
> I think that's because the emoji takes the full height given, but fonts keep some height for ascenders and descenders?

That makes sense.

> I agree, you can have any length string output as long as the first character is an emoji now. So I think it's ok to accept any length string as the symbol type (except for the reserved values e.g. "square"...).
> 
> Then there would be no case for the DIAMOND default. Please update the code and Javadoc accordingly.

Done. Let me know if that's what you had in mind.